ntoskrnl/inbv/inbv.c

```diff
@@ -41,12 +41,34 @@ static INBV_PROGRESS_STATE InbvProgressState;
 static BT_PROGRESS_INDICATOR InbvProgressIndicator = {0, 25, 0};
 static INBV_RESET_DISPLAY_PARAMETERS InbvResetDisplayParameters = NULL;
 static ULONG ResourceCount = 0;
-static PUCHAR ResourceList[1 + IDB_CLUSTER_SERVER]; // First entry == NULL, followed by 'ResourceCount' entries.
+static PUCHAR ResourceList[1 + IDB_MAX_RESOURCE]; // First entry == NULL, followed by 'ResourceCount' entries.
 
 #ifdef INBV_ROTBAR_IMPLEMENTED
+/*
+ * Change this to modify progress bar behaviour
+ */
+#define ROT_BAR_DEFAULT_MODE    RB_PROGRESS_BAR
+
+/*
+ * Values for PltRotBarStatus:
+ * - PltRotBarStatus == 1, do palette fading-in (done elsewhere in ReactOS);
+ * - PltRotBarStatus == 2, do rotation bar animation;
+ * - PltRotBarStatus == 3, stop the animation thread.
+ * - Any other value is ignored and the animation thread continues to run.
+ */
+typedef enum _ROT_BAR_STATUS
+{
+    RBS_FADEIN = 1,
```

Comment:
Is there a reason to make this distinct from 0?

Reply:
Really, why isn't this just a boolean?

Reply:
Originally (and as the original code -- before @binarymaster's patch -- also suggested via the value of "PltRotBarStatus"), this looked like it was designed as a "command" variable for the animation thread to know what to do "next", like, first, do the palette fading-in, then the rotation, then stop (in the usual order, but you can also choose to blank the screen again, then redo a fading in etc...).
In the current code the fading-in is done before starting the animation thread; I was wondering whether it could be nice instead to move it in the animation thread where that PltRotBarStatus variable would then make better sense (I didn't do the move as you can see).
But perhaps we can also do the way of the old bootvid code (see my link above), where the animation thread always do "fading in" first, then bar rotation, and thus the PltRotBarStatus would just become a Boolean "start / stop".

Reply:
I'd prefer leave this variable as is...